lib/astarte_core/mapping.ex

```diff
@@ -221,6 +222,16 @@ defmodule Astarte.Core.Mapping do
     end
   end
 
+  defp check_coherence_database_retention_policy_and_database_retention_ttl(changeset) do
```

Comment:
That's a bit long and could make code hard to read, I'd go for `validate_database_retention_policy_and_ttl` (using `validate` to be consistent with everything else).
This also avoids to leak the implementation detail of what's happening inside the function (i.e. a coherence check) so that if you add/remove checks in that function the name remains valid.